charts/redpanda/templates/secrets.yaml

```
@@ -48,19 +48,23 @@ stringData:
     source /var/lifecycle/common.sh
 
 {{- if gt ( .Values.statefulset.replicas | int64 ) 2 }}
-    set -ex
-
-    until NODE_ID=$(${CURL_NODE_ID_CMD} | grep -o '\"node_id\":[^,}]*' | grep -o '[^: ]*$'); do
-        sleep 0.5
-    done
-
-    echo "Clearing maintenance mode on node ${NODE_ID}"
-    CURL_MAINTENANCE_DELETE_CMD="${CURL_MAINTENANCE_DELETE_CMD_PREFIX} {{ include "admin-tls-curl-flags" . }} ${CURL_URL}/v1/brokers/${NODE_ID}/maintenance"
-    # a 400 here would mean not in maintenance mode
-    until [ "${status:-}" = '"200"' ] || [ "${status:-}" = '"400"' ]; do
-        status=$(${CURL_MAINTENANCE_DELETE_CMD})
-        sleep 0.5
-    done
+    postStartHook () {
+      set -x
+
+      until NODE_ID=$(${CURL_NODE_ID_CMD} | grep -o '\"node_id\":[^,}]*' | grep -o '[^: ]*$'); do
+          sleep 0.5
+      done
+
+      echo "Clearing maintenance mode on node ${NODE_ID}"
+      CURL_MAINTENANCE_DELETE_CMD="${CURL_MAINTENANCE_DELETE_CMD_PREFIX} {{ include "admin-tls-curl-flags" . }} ${CURL_URL}/v1/brokers/${NODE_ID}/maintenance"
+      # a 400 here would mean not in maintenance mode
+      until [ "${status:-}" = '"200"' ] || [ "${status:-}" = '"400"' ]; do
+          status=$(${CURL_MAINTENANCE_DELETE_CMD})
+          sleep 0.5
+      done
+    }
+
+    timeout {{ .Values.statefulset.terminationGracePeriodSeconds }} postStartHook
```

```suggestion
    timeout {{ div .Values.statefulset.terminationGracePeriodSeconds 2 }} postStartHook | true
```
Let's give Redpanda some time after this is terminated. Also never fail or the pod will terminate immediately.